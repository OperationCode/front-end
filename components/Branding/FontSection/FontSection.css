.fontsList {
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
  margin: 0;
  padding: 0;
  list-style: none;
  width: 100%;
}

.fontsList > li {
  margin: 1rem;
}

.primaryFontFamily div > h6,
.primaryFontFamily div > p {
  font-family: var(--primaryFontFamily), sans-serif;
}

.secondaryFontFamily div > h6,
.secondaryFontFamily div > p {
  font-family: var(--secondaryFontFamily), serif;
}
