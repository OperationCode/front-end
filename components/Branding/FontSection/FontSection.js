import React from 'react';
import Section from 'components/_common_/Section/Section';
import { fontsObject } from 'common/styles/styleExports';
import styles from './FontSection.css';

function FontSection() {
  // Every letter of the alphabet in one string
  const demoText = 'Sphinx of black quartz, judge my vow!';

  return (
    <Section title="Fonts" theme="gray">
      <ul className={styles.fontsList}>
        {Object.keys(fontsObject).map(item => {
          const fontStyle = {
            fontFamily: fontsObject[item],
          };
          return (
            <li key={item}>
              <div>
                <h6 style={fontStyle}>{fontsObject[item]}</h6>
                <p style={fontStyle}>{demoText}</p>
              </div>
            </li>
          );
        })}
      </ul>
    </Section>
  );
}

export default FontSection;
