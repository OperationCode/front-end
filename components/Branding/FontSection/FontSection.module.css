.fontsList {
  list-style: none;
  width: 100%;
}

.fontsList > li {
  margin: 1rem;
}

.primaryFontFamily div > h6,
.primaryFontFamily div > p {
  font-family: var(--primaryFontFamily), sans-serif;
}

.secondaryFontFamily div > h6,
.secondaryFontFamily div > p {
  font-family: var(--secondaryFontFamily), serif;
}
