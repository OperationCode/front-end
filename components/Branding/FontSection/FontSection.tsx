import Content from 'components/Content/Content';
import { fontsObject } from 'common/styles/styleExports';

function FontSection() {
  // Every letter of the alphabet in one string
  const demoText = 'Sphinx of black quartz, judge my vow!';

  return (
    <Content
      title="Typography"
      theme="gray"
      hasTitleUnderline
      columns={[
        <ul className="list-none w-full [&>li]:m-4">
          {Object.keys(fontsObject).map(item => {
            const fontStyle = {
              fontFamily: fontsObject[item],
            };
            return (
              <li key={item}>
                <div
                  className={
                    fontsObject[item] === 'DIN Condensed Bold'
                      ? '[&>p]:font-serif [&>h6]:font-serif [&>p]:font-din-condensed [&>h6]:font-din-condensed'
                      : '[&>p]:font-sans [&>p]:font-encode-sans [&>h6]:font-sans [&>h6]:font-encode-sans'
                  }
                >
                  <h6 style={fontStyle}>{fontsObject[item]}</h6>
                  <p style={fontStyle}>{demoText}</p>
                </div>
              </li>
            );
          })}
        </ul>,
      ]}
    />
  );
}

export default FontSection;
