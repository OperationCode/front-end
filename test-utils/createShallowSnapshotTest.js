import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers, components utilizing something that requires large
 * mocking, or very large components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
