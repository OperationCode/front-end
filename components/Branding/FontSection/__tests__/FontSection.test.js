import React from 'react';
import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';

import FontSection from '../FontSection';

describe('FontSection', () => {
  it('should render with required props', () => {
    createShallowSnapshotTest(<FontSection />);
  });
});
