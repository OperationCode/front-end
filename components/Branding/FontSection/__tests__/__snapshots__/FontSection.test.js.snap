// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FontSection should render with required props 1`] = `
<Section
  hasHeadingLines={true}
  id=""
  style={Object {}}
  theme="gray"
  title="Fonts"
>
  <ul
    className="fontsList"
  >
    <li>
      <div>
        <h6
          style={
            Object {
              "fontFamily": "DIN Condensed Bold",
            }
          }
        >
          DIN Condensed Bold
        </h6>
        <p
          style={
            Object {
              "fontFamily": "DIN Condensed Bold",
            }
          }
        >
          Sphinx of black quartz, judge my vow!
        </p>
      </div>
    </li>
    <li>
      <div>
        <h6
          style={
            Object {
              "fontFamily": "Encode Sans",
            }
          }
        >
          Encode Sans
        </h6>
        <p
          style={
            Object {
              "fontFamily": "Encode Sans",
            }
          }
        >
          Sphinx of black quartz, judge my vow!
        </p>
      </div>
    </li>
  </ul>
</Section>
`;
